great_expectations/data_context/data_context.py

```diff
@@ -3611,6 +3627,38 @@ def _test_instantiation_of_data_connector_from_yaml_config(
         )
         return instantiated_class, usage_stats_event_payload
 
+    def _test_instantiation_of_profiler_from_yaml_config(
+        self, name: Optional[str], class_name: str, config: CommentedMap
+    ) -> Tuple[RuleBasedProfiler, dict]:
+        """
+        Helper to create profiler instance and update usage stats payload.
+        See `test_yaml_config` for more details.
+        """
+        print(f"\tInstantiating as a {class_name}, since class_name is {class_name}")
+
+        profiler_name = name or config.get("name") or "my_temp_profiler"
+
+        profiler_config = RuleBasedProfilerConfig.from_commented_map(
+            commented_map=config
+        )
+        profiler_config = profiler_config.to_json_dict()
+        profiler_config.update({"name": profiler_name})
+
+        profiler_class_args: dict = {
+            key: value
+            for key, value in profiler_config.items()
+            if key not in ["module_name", "class_name"]
+        }
```

Comment:
This pattern also doesn't feel great - should we opt for `instantiate_class_from_config` here? That adds more overhead but doesn't require use to get rid of these attributes.

Reply:
Interesting - I can see it both ways, but `instantiate_class_from_config` is how it will eventually be created within GE when using yaml configs so it would be good to use that here for consistency.

Reply:
Agreeing with @anthonyburdi 's comment above

Reply:
refactored to use `instantiate_class_from_config`